### PR TITLE
[CSL-2216] Merge feature/new-wallet into develop (Jan 2018)

### PR DESCRIPTION
**Pull request by adinapoli-iohk**
This quite big PR (I promise, that's the last one) merges `feature/new-wallet` into `develop`. We agreed that from now on we will deliver each new improvement to the wallet (no matter if old or new) in a separate PR, so that we won't end up in this situation again.

This PR is 90% backwards compatible with `develop`, except for two things which needs to be reviewed by @darko-mijic and @domenkozar , but that are easy to fix:

- [x] The `daedalus` folder is gone, as the bridge is not used by Daedalus anymore (owners: Darko);
- [x] The wallet executable has been renamed `cardano-node-old` and the new wallet one has been renamed `cardano-node`. The main idea is to effectively start the deprecation process of the old one in a gradual way by marking explicitly that we are going to stop using the current version of the wallet. **The new wallet executable, despite a slightly different CLI should be totally compatible with the old version, which means it also exposes the old API and can be used by Daedalus as-it-is**. Question has to be answered whether such executable renaming is too pervasive and needs to be reverted (owners: Domen, Darko).

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -116,7 +112,7 @@ after_test:
       }
       .\cache-s3 --prefix=$env:APPVEYOR_PROJECT_NAME --git-branch=$env:APPVEYOR_REPO_BRANCH --suffix=windows -c -v info save stack work
     }
- - xcopy /q /s /e /r /k /i /v /h /y "%WORK_DIR%\daedalus" "%APPVEYOR_BUILD_FOLDER%\daedalus"
+# - xcopy /q /s /e /r /k /i /v /h /y "%WORK_DIR%\daedalus" "%APPVEYOR_BUILD_FOLDER%\daedalus"
```
Comment (domenkozar) [inline appveyor.yml]: Afaik this was needed for the artifacts so it should stay. If not, we should just remove it but not comment it out.

Comment (adinapoli-iohk) [reply appveyor.yml]: Of course it's needed 🤦‍♂️ . Let me take care of that.

**Comment by domenkozar** [comment]
Based on "stability" of the api I suggest we:

1) ship cardano-node-old as part of next release
2) use cardano-node for all development

Assuming these two points, we have to update:

```
$ git grep cardano-node
appveyor.yml:  - copy cardano-node.exe daedalus\
docs/configuration.md:* `cardano-node-simple` and `cardano-node` also have command line option to
docs/how-to/build-cardano-sl-and-daedalus-from-source-code.md:        cardano-node-simple
docs/how-to/build-cardano-sl-and-daedalus-from-source-code.md:   - `cardano-node`, `cardano-swagger`
docs/how-to/build-cardano-sl-and-daedalus-from-source-code.md:be taking window focus until they are finished. If it complains about `cardano-node.log` not existing just
docs/how-to/build-cardano-sl-and-daedalus-from-source-code.md:    [nix-shell:~/daedalus]$ touch ~/.config/Daedalus/Logs/cardano-node.log
docs/how-to/test-update-system.md:Installer should be put by cardano-node to `%APPDATA%\Daedalus\Installer.exe`.
docs/proposals/update-downloader.md:  of writing the update on disk (or cardano-node is stopped). In this
explorer/scripts/launch/staging.sh:    --kademlia-peer cardano-node-0.aws.iohkdev.io:3000 \
explorer/scripts/launch/staging.sh:    --kademlia-peer cardano-node-1.aws.iohkdev.io:3000 \
explorer/scripts/launch/staging.sh:    --kademlia-peer cardano-node-2.aws.iohkdev.io:3000 \
explorer/scripts/launch/staging.sh:    --kademlia-peer cardano-node-3.aws.iohkdev.io:3000 \
explorer/scripts/launch/staging.sh:    --kademlia-peer cardano-node-4.aws.iohkdev.io:3000 \
explorer/scripts/launch/staging.sh:    --kademlia-peer cardano-node-5.aws.iohkdev.io:3000 \
explorer/scripts/launch/staging.sh:    --kademlia-peer cardano-node-6.aws.iohkdev.io:3000 \
explorer/scripts/launch/staging.sh:    --kademlia-peer cardano-node-7.aws.iohkdev.io:3000 \
explorer/scripts/launch/staging.sh:    --kademlia-peer cardano-node-8.aws.iohkdev.io:3000 \
explorer/scripts/launch/staging.sh:    --kademlia-peer cardano-node-9.aws.iohkdev.io:3000 \
explorer/scripts/launch/staging.sh:    --kademlia-peer cardano-node-10.aws.iohkdev.io:3000 \
explorer/scripts/launch/staging.sh:    --kademlia-peer cardano-node-11.aws.iohkdev.io:3000 \
explorer/scripts/launch/staging.sh:    --kademlia-peer cardano-node-12.aws.iohkdev.io:3000 \
explorer/scripts/launch/staging.sh:    --kademlia-peer cardano-node-13.aws.iohkdev.io:3000 \
infra/Pos/Reporting/Methods.hs:                try . sendReport logFile reportType "cardano-node" . toString
lib/src/Pos/Client/CLI/NodeOptions.hs:        ("cardano-node-" <> showVersion version <>
lib/src/Pos/Client/CLI/NodeOptions.hs:  stack exec -- cardano-node                                             \
node/cardano-sl-node.cabal:executable cardano-node-simple
scripts/ci/ci.sh:  cp ../cardano-sl-wallet.root/bin/cardano-node .
scripts/ci/ci.sh:  ./cardano-node --help > /dev/null
scripts/launch/connect-to-cluster/default.nix:    wallet = "${iohkPkgs.cardano-sl-wallet}/bin/cardano-node";
scripts/launch/connect-to-cluster/mainnet-staging.sh:stack exec -- cardano-node                                  \
scripts/launch/demo-with-wallet-api.sh:WALLET_EXE_NAME='cardano-node' WALLET_TEST=1 "$base"/demo.sh $@
scripts/launch/demo.sh:    WALLET_EXE_NAME="cardano-node"
scripts/launch/demo.sh:  exec_name='cardano-node-simple'
scripts/launch/demo.sh:      exec_name='cardano-node'
tools/src/launcher/Main.hs:    --node binaries_v000/cardano-node                       \
tools/src/launcher/Main.hs:        sendReport [txz] (RCrash ec) "cardano-node" reportServ
wallet-new/cardano-sl-wallet-new.cabal:executable cardano-node
wallet-new/server/Cardano/Wallet/Server/CLI.hs:        ("cardano-node-" <> showVersion version <>
wallet/cardano-sl-wallet.cabal:executable cardano-node-old
wallet/node/NodeOptions.hs:        ("cardano-node-" <> showVersion version <>
```

**Comment by domenkozar** [comment]
@adinapoli-iohk +1. Let's keep current name and rename to the new wallet exactly after branching out 1.1

**Comment by adinapoli-iohk** [comment]
@domenkozar I have reverted the commit which was renaming the executables, can I have your blessing now on the CI side? 😉

**Pull request merged**